### PR TITLE
Updated OS and config template entities

### PR DESCRIPTION
**Pull request by sghai**
- Added location for config template entity
- Updated OS entity with update mixin

### CODE REVIEW

**Comment by sghai** [comment]
```
In [5]: os1 = entities.OperatingSystem(server_config).search(query={u'search': u'name="RedHat" AND major="6" AND minor="7"'})[0]
In [7]: os1
Out[7]: nailgun.entities.OperatingSystem(nailgun.config.ServerConfig(url='https://foreman_server', verify=False, auth=('admin', 'changeme')), major=u'6', family=u'Redhat', description=u'RedHat 6.7', release_name=None, id=1, minor=u'7', name=u'RedHat')

In [12]: arch = entities.Architecture(server_config).search(query={u'search': u'name="x86_64"'})[0]
In [14]: arch.id
Out[14]: 1

In [16]: os1.architecture = [arch]
In [17]: os1 = os1.update(['architecture'])

In [18]: os1.architecture
Out[18]: [nailgun.entities.Architecture(nailgun.config.ServerConfig(url='https://foreman_server', verify=False, auth=('admin', 'changeme')), id=1)]

In [19]: arch.id
Out[19]: 1
```

**Comment by sghai** [comment]
`updating template location`

In [6]: org1 = entities.Organization(server_config, name="testorg1").create()
In [7]: loc1 = entities.Location(server_config, name="testloc1", organization=[org1]).create()
In [8]: template1 = entities.ConfigTemplate(server_config).search(query={u'search': u'name="Satellite Kickstart Default"'})[0]
In [14]: template1.location = [loc1]
In [15]: template1 = template1.update(['location'])

In [18]: template1.location
Out[18]: [nailgun.entities.Location(nailgun.config.ServerConfig(url='https://foreman_server', verify=False, auth=('admin', 'changeme')), id=6)]

In [19]: loc1.id
Out[19]: 6

**Comment by oshtaier** [comment]
ACK

File: nailgun/entities.py [hunk 1]
```diff
@@ -2361,6 +2363,25 @@ def create_payload(self):
             u'operatingsystem': super(OperatingSystem, self).create_payload()
         }
 
+    def update(self, fields=None):
```
Comment (Ichimonji10) [inline nailgun/entities.py]: This method is unnecessary.

```python
>>> from nailgun import entities
>>> os = entities.OperatingSystem(id=10).read()
>>> os.name
'127548037869023'
>>> os.name = '474523625643091'
>>> os = os.update(['name'])
>>> os.name
'474523625643091'
```

When we submit a PUT request to Satellite, we expect Satellite to respond with a full description of the updated entity. This method is only necessary if Satellite fails to return a full description.

Comment (Ichimonji10) [inline nailgun/entities.py]: Just to be clear, I dropped the `update` method when creating the sample code above.

**Comment by Ichimonji10** [comment]
ACK pending comment.

**Comment by sghai** [comment]
@Ichimonji10 : Hey.. could you please ACK and merge, if changes looks 'OK' ?

**Comment by Ichimonji10** [comment]
Looks great.

**Comment by elyezer** [comment]
:+1:

**Pull request merged**